File: solutions/kotlin/binary-search/2/src/main/kotlin/BinarySearch.kt
import java.util.NoSuchElementException
object BinarySearch {
    fun search(list: List<Int>, item: Int): Int {
        if (list.isEmpty()) {
            throw NoSuchElementException()
        }
        var left = 0
        var right = list.size - 1
        while (left < right) {
            val middle = (left + right + 1) / 2
            if (list[middle] <= item) {
                left = middle
            } else {
                right = middle - 1
            }
        }
        if (list[left] == item) {
            return left
        } else {
            throw NoSuchElementException()
        }
    }
}

File: solutions/csharp/binary-search/2/BinarySearch.cs
public static class BinarySearch
{
    public static int Find(int[] input, int value)
    {
        if (input.Length == 0)
        {
            return -1;
        }
        var left = 0;
        var right = input.Length - 1;
        while (left < right)
        {
            var middle = (left + right + 1) / 2;
            if (input[middle] <= value)
            {
                left = middle;
            }
            else
            {
                right = middle - 1;
            }
        }
        return input[left] == value ? left : -1;
    }
}